server/sonar-db-dao/src/test/java/org/sonar/db/component/AnalysisPropertiesDaoTest.java

```diff
@@ -0,0 +1,182 @@
+    compareFirstValueWith(analysisPropertyDto);
+  }
+
+  @Test
+  public void insert_as_text() {
+    AnalysisPropertyDto analysisPropertyDto = insertAnalysisPropertyDto(1 + random.nextInt(3999));
+
+    assertThat(dbTester.countRowsOfTable(dbSession, "ANALYSIS_PROPERTIES")).isEqualTo(1);
+    compareFirstValueWith(analysisPropertyDto);
+  }
+
+  @Test
+  public void insert_as_clob() {
+    AnalysisPropertyDto analysisPropertyDto = insertAnalysisPropertyDto(4000 + random.nextInt(100));
+
+    assertThat(dbTester.countRowsOfTable(dbSession, "ANALYSIS_PROPERTIES")).isEqualTo(1);
+    compareFirstValueWith(analysisPropertyDto);
+  }
+
+  @Test
+  public void insert_a_list() {
+    List<AnalysisPropertyDto> propertyDtos = Arrays.asList(
+      getAnalysisPropertyDto(random.nextInt(8000), randomAlphanumeric(40)),
+      getAnalysisPropertyDto(random.nextInt(8000), randomAlphanumeric(40)),
+      getAnalysisPropertyDto(random.nextInt(8000), randomAlphanumeric(40)),
+      getAnalysisPropertyDto(random.nextInt(8000), randomAlphanumeric(40)),
+      getAnalysisPropertyDto(random.nextInt(8000), randomAlphanumeric(40)),
+      getAnalysisPropertyDto(random.nextInt(8000), randomAlphanumeric(40)),
+      getAnalysisPropertyDto(random.nextInt(8000), randomAlphanumeric(40)),
+      getAnalysisPropertyDto(random.nextInt(8000), randomAlphanumeric(40)));
+
+    underTest.insert(dbSession, propertyDtos);
+    assertThat(dbTester.countRowsOfTable(dbSession, "ANALYSIS_PROPERTIES")).isEqualTo(propertyDtos.size());
+  }
+
+  @Test
+  public void selectByAnalysisUuid_should_return_correct_values() {
+    String snapshotUuid = randomAlphanumeric(40);
+
+    List<AnalysisPropertyDto> propertyDtos = Arrays.asList(
+      getAnalysisPropertyDto(random.nextInt(8000), snapshotUuid),
+      getAnalysisPropertyDto(random.nextInt(8000), snapshotUuid),
+      getAnalysisPropertyDto(random.nextInt(8000), snapshotUuid),
+      getAnalysisPropertyDto(random.nextInt(8000), snapshotUuid),
+      getAnalysisPropertyDto(random.nextInt(8000), snapshotUuid),
+      getAnalysisPropertyDto(random.nextInt(8000), snapshotUuid),
+      getAnalysisPropertyDto(random.nextInt(8000), snapshotUuid),
+      getAnalysisPropertyDto(random.nextInt(8000), snapshotUuid));
+
+    underTest.insert(dbSession, propertyDtos);
+    assertThat(dbTester.countRowsOfTable(dbSession, "ANALYSIS_PROPERTIES")).isEqualTo(propertyDtos.size());
+
+    List<AnalysisPropertyDto> result = underTest.selectBySnapshotUuid(dbSession, snapshotUuid);
+    assertThat(result).containsExactlyInAnyOrder((AnalysisPropertyDto[]) propertyDtos.toArray());
+  }
+
+  private AnalysisPropertyDto insertAnalysisPropertyDto(int valueLength) {
+    AnalysisPropertyDto analysisPropertyDto = getAnalysisPropertyDto(valueLength, randomAlphanumeric(40));
+    underTest.insert(dbSession, analysisPropertyDto);
+    return analysisPropertyDto;
+  }
+
+  private AnalysisPropertyDto getAnalysisPropertyDto(int valueLength, String snapshotUuid) {
```

Comment:
I suggest to rename this method to `newAnalysisPropertyDto` or `createAnalysisPropertyDto`. I had to check it's implementation to be sure what the method did when used in test `selectByAnalysisUuid_should_return_correct_values`. `get` really suggest DTO is retrieved from DB.